passage/src/main/java/id/passage/android/CurrentUser.kt

```
@@ -226,12 +137,101 @@ final class PassageUser private constructor(
+     *
+     * Retrieves social connections for the current user. The user must be authenticated via a bearer token.
+     * @return SocialConnectionsResponse containing the list of social connections
+     * @throws PassageUserException If an error occurs during the retrieval process
+     */
+    suspend fun socialConnections(): UserSocialConnections {
+        val currentUserAPI = CurrentuserAPI(Passage.BASE_PATH)
+        try {
+            return currentUserAPI.getCurrentuserSocialConnections(Passage.appId).socialConnections
+        } catch (e: Exception) {
+            throw PassageUserException.convert(e)
+        }
+    }
+
+    /**
+     * Delete SocialConnection
+     *
+     * Deletes a social connection for the current user. The user must be authenticated via a bearer token.
+     * @param socialConnectionType The type of social connection to delete
+     * @throws PassageUserException If an error occurs during the deletion process
+     */
+    suspend fun deleteSocialConnection(socialConnectionType: SocialConnection) {
+        val currentUserAPI = CurrentuserAPI(Passage.BASE_PATH)
+        try {
+            return currentUserAPI.deleteCurrentuserSocialConnection(Passage.appId, socialConnectionType)
+        } catch (e: Exception) {
+            throw PassageUserException.convert(e)
+        }
+    }
+
+    /**
+     * Retrieve Current User Metadata
+     *
+     * Fetches the metadata associated with the current user. This method requires that the user is authenticated via a bearer token.
+     * The metadata contains various attributes associated with the user, which can be used for user-specific operations.
+     * @return Metadata? containing the current user's metadata, null if retrieval fails
+     */
+    suspend fun metadata(): Metadata? {
+        val currentUserAPI = CurrentuserAPI(Passage.BASE_PATH)
+        return try {
+            currentUserAPI.getCurrentuserMetadata(Passage.appId)
+        } catch (e: Exception) {
+            null
```

I think we want to throw an exception here.